### PR TITLE
ikev1: add ikev1 parser - v7

### PR DESCRIPTION
**Pull request by fhonza**
Previous PR: https://github.com/OISF/suricata/pull/5353

Describe changes:
- added struct AppLayerProtoDetectAliases to register aliases for protocols
- updates to documentation

suricata-verify-pr: 324
[324](https://github.com/OISF/suricata-verify/pull/324)

### CODE REVIEW

**Comment by fhonza** [comment]
needed to modify suricata-verify, the PR is now here https://github.com/OISF/suricata-verify/pull/324.

hope that's ok, didn't want to create new PR for the link....

**Review by catenacyber — CHANGES_REQUESTED** [review]
Thanks for this work.

CI is not happy :-(

Could you review your commits so that each one compiles ?

And have them formatted ok with `git rebase master -x "git clang-format HEAD^"` ?

File: doc/userguide/output/eve/eve-json-format.rst [hunk 1]
```diff
@@ -1804,3 +1804,111 @@ Example of HTTP2 logging, of a request and response:
...
+
+Examples
+~~~~~~~~
+
+Example of IKE logging:
+
+::
+
+  "ike": {
+    "init_spi": "e47a591fd057587f",
```
Comment (catenacyber) [inline doc/userguide/output/eve/eve-json-format.rst:1841]: This documentation seems inconsistent with Suricata-Verify test cf https://github.com/OISF/suricata-verify/pull/324/files#diff-934bf1defac0138e1aaeb5add9dc8bbbR19 where we use `ikev1.init_spi` or `ikev2.init_spi`...

What is the good way to go ?

**Comment by catenacyber** [comment]
The commit

> ikev1: add comparison-mode LTE/GTE for Detect(U32/u8)Data

should rather be named

> detect: add comparison-mode LTE/GTE for Detect(U32/u8)Data

As it is generic and not only for ikev1, right ?

**Comment by catenacyber** [comment]
Same comment for https://github.com/OISF/suricata/pull/5375/commits/56864a09cef23c205a0b1bd2415417cec6be4c5b

I think it should be `util` or `rust` rather than `ikev1`

**Comment by catenacyber** [comment]
> needed to modify suricata-verify, the PR is now here OISF/suricata-verify#324.
> hope that's ok, didn't want to create new PR for the link....

I edited your first message 
The bash magic from .github/workflows/builds.yml  is 
`sv_pr=$(echo "${body}" | awk '/^suricata-verify-pr/ { print $2 }')`
So I guess you cannot use the link

File: doc/userguide/output/eve/eve-json-format.rst (hunk 1 above)
Comment (fhonza) [reply doc/userguide/output/eve/eve-json-format.rst:1841]: it should be ikev1, forgot to update documentation there....

**Comment by fhonza** [comment]
> The commit
> 
> > ikev1: add comparison-mode LTE/GTE for Detect(U32/u8)Data
> 
> should rather be named
> 
> > detect: add comparison-mode LTE/GTE for Detect(U32/u8)Data
> 
> As it is generic and not only for ikev1, right ?

will update the commits, was not sure about the "correct" prefix

**Pull request closed without merging**